sdk/Lusid.Sdk/Utilities/LusidApiFactoryBuilder.cs

```diff
@@ -17,14 +19,15 @@ public static ILusidApiFactory Build(string apiSecretsFilename)
         /// <summary>
         /// Create an ILusidApiFactory using the specified Url and Token Provider
         /// </summary>
-        public static ILusidApiFactory Build(string url, ITokenProvider tokenProvider)
+        public static ILusidApiFactory Build(string url, ITokenProvider tokenProvider, Dictionary<string, string> defaultHeaders = null)
         {
             // TokenProviderConfiguration.ApiClient is the client used by LusidApiFactory and is 
             // NOT thread-safe, so there needs to be a separate instance for each instance of LusidApiFactory.
             // Do NOT cache the LusidApiFactory instances (DEV-6922)
             var config = new TokenProviderConfiguration(tokenProvider)
             {
-                BasePath = url
+                BasePath = url,
+                DefaultHeaders = defaultHeaders
```

Comment:
Maybe only set this if the value passed in is not null, it _might_ be possible that the base class adds in some defaults. Perhaps also add/merge the dictionary rather replace

Reply:
Yeah I thought about these. I ran some of the other tests to see if there were any default headers, and it was just a `null` object. I'll only assign if not null, but I dont think the merge is needed?

Reply:
yeah, that works